tests/conftest.py

```diff
@@ -234,6 +235,32 @@ def pytest_collection_modifyitems(config, items):
                 item.add_marker(skip_marker)
 
 
+def pytest_runtest_makereport(item, call):
+    """Decorate test results with additional properties."""
+    if call.when == "setup":
+        for prop_name, prop_val in GLOBAL_PROPS.items():
+            # if record_testsuite_property worked with xdist we could use that
+            # https://docs.pytest.org/en/7.1.x/reference/reference.html#record-testsuite-property
+            # to record the properties once per report. But here we record each
+            # prop per test. It just results in larger report files.
+            item.user_properties.append((prop_name, prop_val))
+
+        function_docstring = inspect.getdoc(item.function)
+        description = []
+        attributes = {}
+        for line in function_docstring.split("\n"):
+            # extract tags like @type, @issue, etc
+            match = re.match(r"\s*@(?P<attr>\w+):\s*(?P<value>\w+)", line)
+            if match is not None:
+                attr, value = match["attr"], match["value"]
+                attributes[attr] = value
+            else:
+                description.append(line)
```

Comment:
Nit: Can flip cases to avoid not.
```suggestion
            if match is None:
                description.append(line)
            else:
                attr, value = match["attr"], match["value"]
                attributes[attr] = value
```

Reply:
Can just do `if match` as `None` is falsy